texar/tf/utils/beam_search.py

```diff
@@ -41,6 +42,8 @@ def _merge_beam_dim(tensor):
     Returns:
         Reshaped tensor of shape [A*B, ...]
     """
+    if not isinstance(tensor, tf.Tensor):
+        return tensor
```

Comment:
Why directly return `tensor` here instead of converting it into a `tf.Tensor` and continue the subsequent operations?

Reply:
When we implement beam search decoding in `AttentionRNNDecoder` by using this function, `state` used in this function is `AttentionWrapperState` which contains `int` type attribute `time`. This check is used to only reshape the correct attributes in `state`. We also have similar checking in `texar-pytorch`.

Reply:
Thanks for explaining. Can you add brief comments to make the code more readable? E.g., "if `tensor` is not tf.Tensor, then `tensor` is xxx, return directly".